app/frontend/actions/delegate.ts

```diff
@@ -64,59 +70,64 @@ export default (store: Store) => {
     const poolHash = state.shelleyDelegation?.selectedPool?.poolHash as string
     const isStakingKeyRegistered = getSourceAccountInfo(state).shelleyAccountInfo.hasStakingKey
     const stakingAddress = getSourceAccountInfo(state).stakingAddress
-    const txPlanResult = await prepareTxPlan({
-      poolHash,
-      stakingAddress,
-      isStakingKeyRegistered,
-      txType: TxType.DELEGATE,
-    })
-    const newState = getState()
-    if (hasPoolIdentifiersChanged(newState)) {
-      return
-    }
-    const balance = getSourceAccountInfo(state).balance as Lovelace
-
-    if (txPlanResult.success === true) {
-      setState({
-        shelleyDelegation: {
-          ...newState.shelleyDelegation,
-          delegationFee: (txPlanResult.txPlan.fee + txPlanResult.txPlan.deposit) as Lovelace,
-        },
+    if (stakingAddress && poolHash) {
+      const txPlanResult = await prepareTxPlan({
+        poolHash,
+        stakingAddress,
+        isStakingKeyRegistered,
+        txType: TxType.DELEGATE,
       })
-      const delegationTransactionSummary: DelegateTransactionSummary = {
-        type: TxType.DELEGATE,
-        deposit: txPlanResult.txPlan.deposit,
-        stakePool: newState.shelleyDelegation?.selectedPool,
+      const newState = getState()
+      if (hasPoolIdentifiersChanged(newState)) {
+        return
+      }
+      const balance = getSourceAccountInfo(state).balance as Lovelace
+
+      if (isTxPlanResultSuccess(txPlanResult)) {
```

Comment:
My comment https://github.com/vacuumlabs/adalite/pull/1192#discussion_r741082442 referred to all other instances of this helper as well

Reply:
This is doing more then just abstracting the logic to check for the `.success === true`. It is narrowing the Union so that `txPlanResult` inside the conditional is assigned the success type and outside the conditional is assigned the error type. This is useful because it makes accessing different properties cleaner based on the narrowing.

I am happy to change it back but I believe that type narrowing is useful here.